pkg/template/api/validation/validation.go

```diff
@@ -51,3 +55,116 @@ func validateTemplateBody(template *api.Template) (allErrs field.ErrorList) {
 	allErrs = append(allErrs, unversionedvalidation.ValidateLabels(template.ObjectLabels, field.NewPath("labels"))...)
 	return
 }
+
+// ValidateTemplate tests if required fields in the Template are set.
+func ValidateTemplateInstance(oc *client.Client, ctx kapi.Context, ti *api.TemplateInstance) (allErrs field.ErrorList) {
```

Comment:
Maybe this is different than what I'm expecting, but I expect everything in this package to be related to static validation of an object.

Validation packages cannot rely on clients or context.  This looks a lot like an admission plugin.